src/librustc/traits/on_unimplemented.rs

```diff
@@ -22,17 +22,19 @@ pub struct OnUnimplementedDirective {
     pub message: Option<OnUnimplementedFormatString>,
     pub label: Option<OnUnimplementedFormatString>,
     pub note: Option<OnUnimplementedFormatString>,
+    pub enclosing_scope: Option<OnUnimplementedFormatString>,
 }
 
 pub struct OnUnimplementedNote {
     pub message: Option<String>,
     pub label: Option<String>,
     pub note: Option<String>,
+    pub enclosing_scope: Option<String>,
 }
 
 impl OnUnimplementedNote {
     pub fn empty() -> Self {
```

Comment:
Pre-existing, but perhaps we should just implement `Default` here?

Reply:
Seems reasonable, I'll add it